### PR TITLE
feat: add overrides support to bucketing and add tests

### PR DESCRIPTION
**Pull request by ajwootto**
- support overrides in both JS and WASM libs for bucketing
- add additional optional overrides param to JS
- add new method with overrides param to WASM lib
- tests

### CODE REVIEW

File: lib/shared/bucketing-assembly-script/assembly/managers/eventQueueManager.ts [hunk 1]
```diff
@@ -121,6 +121,7 @@ export function queueEvent(
         _getConfigData(sdkKey),
         dvcUser,
         _getClientCustomData(sdkKey),
+        null,
```
Comment (kaushalkapasi) [inline lib/shared/bucketing-assembly-script/assembly/managers/eventQueueManager.ts:124]: if this is always `null` does that mean the `feature: variation` ids will be skipped from the featureVars map for overrides / self-targeting features?

Comment (ajwootto) [reply lib/shared/bucketing-assembly-script/assembly/managers/eventQueueManager.ts:124]: technically yeah, but the event queue logic is only used by serverside SDKs, which currently would not deal with overrides directly. The only thing that matters is the code used by the bucketing API, which is just the config generator  portion.

**Review by kaushalkapasi — APPROVED** [review]

File: lib/shared/bucketing-assembly-script/assembly/bucketing/bucketing.ts [hunk 2]
```diff
@@ -196,38 +229,77 @@ export function bucketUserForVariation(
...
+
+            if (variation == null) {
+                continue
+            }
+
+            bucketedFeatureVariations.push({ feature, variation })
+        }
+    }
+
+    for (let i = 0; i < bucketedFeatureVariations.length; i++) {
```
Comment (jonathannorris) [inline lib/shared/bucketing-assembly-script/assembly/bucketing/bucketing.ts:286]: Ideally, we could structure this to not have to write to `bucketedFeatureVariations` and iterate over it again, as we are adding some allocations and needing to loop over this new array.

Comment (ajwootto) [reply lib/shared/bucketing-assembly-script/assembly/bucketing/bucketing.ts]: updated

File: lib/shared/bucketing-assembly-script/assembly/types/configBody.ts [hunk 3]
```diff
@@ -89,64 +88,107 @@ export class ConfigBody {
...
+    static fromString(
+        configStr: string,
+        etag: string | null = null,
+    ): ConfigBody {
         const configJSON = JSON.parse(configStr)
-        if (!configJSON.isObj) throw new Error('generateBucketedConfig config param not a JSON Object')
+        if (!configJSON.isObj)
+            throw new Error(
+                'generateBucketedConfig config param not a JSON Object',
+            )
```
Comment (jonathannorris) [inline lib/shared/bucketing-assembly-script/assembly/types/configBody.ts:114]:
```suggestion
        if (!configJSON.isObj) {
            throw new Error(
                'generateBucketedConfig config param not a JSON Object',
            )
          }
```

Comment (jonathannorris) [inline lib/shared/bucketing-assembly-script/assembly/types/configBody.ts:114]:
```suggestion
        if (!configJSON.isObj) {
            throw new Error(
                'generateBucketedConfig config param not a JSON Object',
            )
        }
```

File: lib/shared/bucketing-assembly-script/assembly/types/configBody.ts [hunk 4]
```diff
@@ -184,4 +226,10 @@ export class ConfigBody {
             ? this._variableIdToFeatureMap.get(variable_id)
             : null
     }
+
+    getFeatureForId(feature_id: string): Feature | null {
```
Comment (jonathannorris) [inline lib/shared/bucketing-assembly-script/assembly/types/configBody.ts:230]: What is this used for?

**Review by jonathannorris — CHANGES_REQUESTED** [review]

**Review by jonathannorris — APPROVED** [review]

**Pull request merged**